File: src/classes/api/req_res/get-shared-kyous-response.ts
'use strict'

import { Kyou } from '@/classes/datas/kyou'
import { GkillAPIResponse } from '../gkill-api-response'
import type { Text } from '@/classes/datas/text'
import type { Mi } from '@/classes/datas/mi'
import type { Tag } from '@/classes/datas/tag'
import { TimeIs } from '@/classes/datas/time-is'
import type { Kmemo } from '@/classes/datas/kmemo'
import type { KC } from '@/classes/datas/kc'
import type { Nlog } from '@/classes/datas/nlog'
import type { Lantana } from '@/classes/datas/lantana'
import type { URLog } from '@/classes/datas/ur-log'
import type { IDFKyou } from '@/classes/datas/idf-kyou'
import type { ReKyou } from '@/classes/datas/re-kyou'
import type { GitCommitLog } from '@/classes/datas/git-commit-log'
import type { GPSLog } from '@/classes/datas/gps-log'

export class GetSharedKyousResponse extends GkillAPIResponse {
    title: string
    view_type: string
    kyous: Array<Kyou>
    kmemos: Array<Kmemo>
    kcs: Array<KC>
    mis: Array<Mi>
    nlogs: Array<Nlog>
    lantanas: Array<Lantana>
    timeiss: Array<TimeIs>
    urlogs: Array<URLog>
    idf_kyous: Array<IDFKyou>
    rekyous: Array<ReKyou>
    git_commit_logs: Array<GitCommitLog>
    gps_logs: Array<GPSLog>
    attached_tags: Array<Tag>
    attached_texts: Array<Text>
    attached_timeiss: Array<TimeIs>

    constructor() {
        super()
        this.title = ""
        this.view_type = "rykv"
        this.kyous = new Array<Kyou>()
        this.kmemos = new Array<Kmemo>()
        this.kcs = new Array<KC>()
        this.mis = new Array<Mi>()
        this.nlogs = new Array<Nlog>()
        this.lantanas = new Array<Lantana>()
        this.timeiss = new Array<TimeIs>()
        this.urlogs = new Array<URLog>()
        this.idf_kyous = new Array<IDFKyou>()
        this.rekyous = new Array<ReKyou>()
        this.git_commit_logs = new Array<GitCommitLog>()
        this.gps_logs = new Array<GPSLog>()
        this.attached_tags = new Array<Tag>()
        this.attached_texts = new Array<Text>()
        this.attached_timeiss = new Array<TimeIs>()
    }
}

File: src/app/gkill/api/req_res/get_shared_kyous_response.go
package req_res

import (
	"github.com/mt3hr/gkill/src/app/gkill/api/message"
	"github.com/mt3hr/gkill/src/app/gkill/dao/reps"
)

type GetSharedKyousResponse struct {
	Messages []*message.GkillMessage `json:"messages"`

	Errors []*message.GkillError `json:"errors"`

	Title string `json:"title"`

	ViewType string `json:"view_type"`

	Kyous []*reps.Kyou `json:"kyous"`

	Kmemos []*reps.Kmemo `json:"kmemos"`

	KCs []*reps.KC `json:"kcs"`

	Mis []*reps.Mi `json:"mis"`

	Nlogs []*reps.Nlog `json:"nlogs"`

	Lantanas []*reps.Lantana `json:"lantanas"`

	URLogs []*reps.URLog `json:"urlogs"`

	IDFKyous []*reps.IDFKyou `json:"idf_kyous"`

	ReKyous []*reps.ReKyou `json:"rekyous"`

	GitCommitLogs []*reps.GitCommitLog `json:"git_commit_logs"`

	GPSLogs []*reps.GPSLog `json:"gps_logs"`

	AttachedTags []*reps.Tag `json:"attached_tags"`

	AttachedTexts []*reps.Text `json:"attached_texts"`

	AttachedTimeIss []*reps.TimeIs `json:"attached_timeiss"`
}
